src/core/Image.tsx

```diff
@@ -92,12 +106,62 @@ export const Image = React.forwardRef(
           map-encoding={gl.outputEncoding}
           zoom={zoom}
           grayscale={grayscale}
+          opacity={opacity}
+          scale={planeBounds}
+          imageBounds={imageBounds}
+          toneMapped={toneMapped}
+          transparent={transparent}
+        />
+        {children}
+      </mesh>
+    )
+  }
+)
+
+const ImageWithTexture = React.forwardRef(
+  (
+    {
+      children,
+      color,
+      segments = 1,
+      scale = 1,
+      zoom = 1,
+      grayscale = 0,
+      opacity = 1,
+      texture,
+      toneMapped,
+      transparent,
+      ...props
+    }: ImageProps,
+    ref: React.ForwardedRef<THREE.Mesh>
+  ) => {
+    extend({ ImageMaterial: ImageMaterialImpl })
+    const gl = useThree((state) => state.gl)
+    const planeBounds = Array.isArray(scale) ? [scale[0], scale[1]] : [scale, scale]
+    const imageBounds = [texture!.image.width, texture!.image.height]
+    return (
+      <mesh ref={ref} scale={scale} {...props}>
+        <planeGeometry args={[1, 1, segments, segments]} />
+        <imageMaterial
+          color={color}
+          map={texture!}
+          map-encoding={gl.outputEncoding}
+          zoom={zoom}
+          grayscale={grayscale}
+          opacity={opacity}
           scale={planeBounds}
           imageBounds={imageBounds}
           toneMapped={toneMapped}
+          transparent={transparent}
         />
         {children}
       </mesh>
     )
   }
 )
+
+export const Image = React.forwardRef<THREE.Mesh, ImageProps>((props, ref) => {
+  if (props.url) return <ImageWithUrl {...props} ref={ref} />
+  else if (props.texture) return <ImageWithTexture {...props} ref={ref} />
+  else throw new Error('<Image /> requires a url or texture')
```

Comment:
Is this anti-pattern? Throwing an error in a component like that

Reply:
not that i know of. it will end up in an error boundary.